packages/@vuepress/bundler-vite/src/plugin/createConstantReplacementPlugin.ts

```diff
@@ -0,0 +1,53 @@
+import type { Plugin } from 'vite'
+
+const fileRegex = /\.html\.vue$/
+const templateRegex = /<template>[\s\S]*<\/template>/g
+
+export const createConstantReplacementPlugin = (): Plugin => {
+  let defineRegex: RegExp | null = null
+
+  return {
+    name: 'vuepress:constant-replacement',
+    enforce: 'pre',
+    apply: 'build',
+    configResolved(resolvedConfig) {
+      if (resolvedConfig.define) {
+        const define = Reflect.ownKeys(resolvedConfig.define) as string[]
+        defineRegex = new RegExp(
+          `\\b(${define
+            .map((key) => key.replace(/[-[\]/{}()*+?.\\^$|]/g, '\\$&'))
+            .join('|')})`,
+          'g'
+        )
+      }
+    },
+    transform(code, id) {
+      if (fileRegex.test(id)) {
+        for (const match of code.matchAll(templateRegex)) {
```

Comment:
`@vue/compiler-sfc` will only recognize one template block. So let's remove the `g` flag and remove the for loop